datastreams/context.go

```diff
@@ -11,6 +11,14 @@ import (
 
 type contextKey struct{}
 
+type CheckpointParams struct {
+	PayloadSize int64
+}
+
+func NewCheckpointParams(payloadSize int64) CheckpointParams {
```

Comment:
i think having this public function may cause the same issue where if we add new features we'll need to update this function (which isn't backwards compatible for clients) or add a new one. maybe we can just remove it and let clients construct the CheckpointParams on their own and only fill in the params they need? Or we can use a Builder pattern?

Reply:
Yeah, I was thinking of using the builder patter in the future, if we need to add more parameters.